control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -391,39 +407,212 @@ func (r Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkainterna
+	if err != nil {
+		return fmt.Errorf("error getting client sarama config: %w", err)
+	}
+
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])
 
-	kafkaClient, err := r.NewKafkaClient(bootstrapServers, saramaConfig)
+	kafkaClusterAdminClient, err := r.NewKafkaClusterAdminClient(bootstrapServers, kafkaClusterAdminSaramaConfig)
 	if err != nil {
-		return fmt.Errorf("failed to create Kafka cluster client: %w", err)
+		return fmt.Errorf("cannot obtain Kafka cluster admin, %w", err)
 	}
-	defer kafkaClient.Close()
+	defer kafkaClusterAdminClient.Close()
 
-	kafkaAdminClient, err := sarama.NewClusterAdminFromClient(kafkaClient)
+	kafkaClient, err := r.NewKafkaClient(bootstrapServers, kafkaClientSaramaConfig)
 	if err != nil {
-		return fmt.Errorf("failed to create Kafka cluster admin client: %w", err)
+		return fmt.Errorf("failed to create Kafka cluster client: %w", err)
 	}
-	defer kafkaAdminClient.Close()
+	defer kafkaClient.Close()
 
 	groupId := cg.Spec.Template.Spec.Configs.Configs["group.id"]
 	topics := cg.Spec.Template.Spec.Topics
 
-	if _, err := r.InitOffsetsFunc(ctx, kafkaClient, kafkaAdminClient, topics, groupId); err != nil {
+	if _, err := r.InitOffsetsFunc(ctx, kafkaClient, kafkaClusterAdminClient, topics, groupId); err != nil {
 		return fmt.Errorf("failed to initialize offset: %w", err)
 	}
 
 	return nil
 }
 
+func (r Reconciler) reconcileKedaObjects(ctx context.Context, cg *kafkainternals.ConsumerGroup) error {
+	var triggerAuthentication *kedav1alpha1.TriggerAuthentication
+	var secret *corev1.Secret
+
+	if hasAuthSpecAuthConfig(cg.Spec.Template.Spec.Auth) || hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) {
+		saslType, err := r.retrieveSaslTypeIfPresent(ctx, cg)
+		if err != nil {
+			return err
+		}
+
+		triggerAuthentication, secret, err = kedafunc.GenerateTriggerAuthentication(cg, saslType)
+		if err != nil {
+			return err
+		}
+	}
+
+	triggers, err := kedafunc.GenerateScaleTriggers(cg, triggerAuthentication)
+	if err != nil {
+		return err
+	}
+
+	scaledObject, err := keda.GenerateScaledObject(cg, cg.GetGroupVersionKind(), kedafunc.GenerateScaleTarget(cg), triggers)
+	if err != nil {
+		return err
+	}
+
+	if triggerAuthentication != nil && secret != nil {
+		err = r.reconcileSecret(ctx, secret, cg)
+
+		// if the event was wrapped inside an error, consider the reconciliation as failed
+		if _, isEvent := err.(*reconciler.ReconcilerEvent); !isEvent {
+			return err
+		}
+
+		err = r.reconcileTriggerAuthentication(ctx, triggerAuthentication, cg)
+
+		// if the event was wrapped inside an error, consider the reconciliation as failed
+		if _, isEvent := err.(*reconciler.ReconcilerEvent); !isEvent {
+			return err
+		}
+	}
+
+	err = r.reconcileScaledObject(ctx, scaledObject, cg)
+
+	// if the event was wrapped inside an error, consider the reconciliation as failed
+	if _, isEvent := err.(*reconciler.ReconcilerEvent); !isEvent {
+		return err
+	}
```

Comment:
The error handling here is weird, we generate a events inside various functions but then we discard it.

I think we should generate events when we do a positive action like "Scaled object updated" or "Scaled object created" but then if there is an error we return an error without returning an event and let the reconciler generate that error event based on the error message.